### ISSUE TITLE
?fix: incredibly low loss after 4 epochs

### ISSUE THREAD

**Issue by MartinBernstorff** [Aarhus-Psychiatry-Research/psycop-common#721]
![image](https://github.com/Aarhus-Psychiatry-Research/psycop-common/assets/8526086/1ca1ca36-67ae-429e-894b-ae3328480892)
(Steps on x-axis)

Requires VPN:
http://exrhel0371.it.rm.dk:5050/#/metric/val_loss?runs=[%22dce1b28c9b1a4bc299b21aaf723ec32d%22]&experiments=[%22598237506833910234%22]&plot_metric_keys=[%22val_loss%22]&plot_layout={%22autosize%22:true,%22xaxis%22:{%22autorange%22:true,%22type%22:%22linear%22},%22yaxis%22:{%22type%22:%22log%22,%22autorange%22:true,%22exponentformat%22:%22e%22}}&x_axis=step&y_axis_scale=linear&line_smoothness=1&show_point=false&deselected_curves=[]&last_linear_y_axis_range=[]

Hypotheses:
* Padding tokens are used for generating loss
* Extremely homogenous sequences – they are pretty homogenous, but not entirely. Seems like a relatively likely explanation.

@KennethEnevoldsen, any thoughts here? :-)

**Comment by KennethEnevoldsen** [comment]
> Padding tokens are used for generating loss

You can check, but I believe the loss ignores the padding token id (-1 I believe). It at leasts shouldn't be the case.

> Extremely homogenous sequences – they are pretty homogenous, but not entirely. Seems like a relatively likely explanation.

Def. a more likely explanation.

**Comment by MartinBernstorff** [comment]
> You can check, but I believe the loss ignores the padding token id (-1 I believe). It at leasts shouldn't be the case.

We've tried looking at the `test_train` test. When within the `PretrainerBEHRT` scope, we get:

* Inputs has two "is_padding" tokens

We then tried looking at the logits here:

https://github.com/Aarhus-Psychiatry-Research/psycop-common/blob/5eead4430299050fde72417a8e3f5220f737cffb/psycop/common/sequence_models/tasks/pretrainer_behrt.py#L81

But none of them are "-1". Does this imply that it does not ignore padding when calculating logits?

I'm way out of my depth here, so it might make sense for you to be hands on?
